packages/onboarding/src/step-container-v2/components/ContentWrapper/ContentWrapper.tsx

```diff
@@ -1,29 +1,48 @@
 import clsx from 'clsx';
-import { ReactNode } from 'react';
+import { ReactNode, Children, isValidElement } from 'react';
 
 import './style.scss';
 
-export const ContentWrapper = ( {
+type MaxWidth = 'wide' | 'huge' | 'xhuge';
+
+const ContentWrapper = ( {
 	children,
-	width,
+	maxWidth = 'wide',
 	centerAligned,
 	hasPadding = true,
 }: {
 	children: ReactNode;
-	width?: 'wide' | 'full';
+	maxWidth?: MaxWidth;
 	centerAligned?: boolean;
 	hasPadding?: boolean;
 } ) => {
+	const childrenCount = Children.toArray( children ).filter( isValidElement ).length;
+
 	return (
 		<div
-			className={ clsx( 'step-container-v2__content-wrapper', {
-				wide: width === 'wide',
-				full: width === 'full',
+			className={ clsx( 'step-container-v2__content-wrapper', maxWidth, {
 				'center-aligned': centerAligned,
 				padding: hasPadding,
 			} ) }
+			style={ { gridTemplateRows: `repeat( ${ childrenCount }, min-content )` } }
 		>
 			{ children }
 		</div>
 	);
 };
+
+ContentWrapper.Row = ( { columns, children }: { columns: number; children: ReactNode } ) => {
+	return (
+		<div
+			className="step-container-v2__content-wrapper-row"
+			style={ {
+				// @ts-expect-error -- This is a valid CSS variable.
+				'--columns': columns,
+			} }
```

Comment:
`as` is better in this situation because `@ts-expect-error` is more powerful than we need. It ignores things like spelling mistakes but we don't need something that powerful.

```suggestion
			style={ {
				'--columns': columns,
			} as React.CSSProperties }
```